### PR TITLE
Doc/general docstring edits

### PR DESCRIPTION
**Pull request by PipKat**
Pre-lunch edits,

### CODE REVIEW

File: pyaedt/modules/SetupTemplates.py [hunk 1]
```diff
@@ -1057,7 +1058,7 @@ def __init__(self, oanalysis, setupname, sweepname, sweeptype="Interpolating", p
         if props:
             self.props = props
         else:
-            self.props["Type"] = sweeptype
+            self.props["Type"] = sweeptypeInitializes, creates, and updates sweeps in HFSS
```
Comment (PipKat) [inline pyaedt/modules/SetupTemplates.py:1061]:
```suggestion
            self.props["Type"] = sweeptype
```

File: pyaedt/edb_core/siwave.py [hunk 2]
```diff
@@ -156,18 +174,21 @@ def phase(self,value):
...
     def impedance(self):
+        """Impedance."""
         return self._impedance
 
     @impedance.setter
     def impedance(self,value):
         self._impedance = value
 
     @property
+    """Property."""
```
Comment (PipKat) [inline pyaedt/edb_core/siwave.py:185]:
```

Comment (PipKat) [inline pyaedt/edb_core/siwave.py:185]:
```suggestion
```

File: pyaedt/edb_core/siwave.py [hunk 3]
```diff
@@ -156,18 +174,21 @@ def phase(self,value):
...
     def source_type(self):
```
Comment (PipKat) [inline pyaedt/edb_core/siwave.py:186]:
```suggestion
    def source_type(self):
    """Property."""
```

File: pyaedt/edb_core/siwave.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 """
-This module contains the ``EdbSiwave`` class.
+This module contains these clases: `CircuitPort`, `CurrentSource`, `EdbSiwave`, 
+`PinGroup`, `ResistorSource`, `Source`, `SourceType`, and VoltageSource.
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:3]:
```suggestion
`PinGroup`, `ResistorSource`, `Source`, `SourceType`, and `VoltageSource`.
```
Should VoltageSource be wrapped in tick marks?

File: pyaedt/edb_core/siwave.py [hunk 5]
```diff
@@ -112,13 +124,15 @@ def do_pin_grouping(self,value):
 
 
 class CircuitPort(Source):
+    """Manages circuit ports."""
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:127]:
```suggestion
    """Manages a circuit port."""
```
To be consistent with the other boundary classes.

File: pyaedt/edb_core/siwave.py [hunk 6]
```diff
@@ -164,10 +183,12 @@ def impedance(self,value):
 
     @property
     def source_type(self):
+        """Property."""
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:186]:
```suggestion
        """Source type."""
```
Is the description `Source type` more accurate?

File: pyaedt/edb_core/siwave.py [hunk 7]
```diff
@@ -220,12 +247,18 @@ def rvalue(self, value):
...
         return self.source_type
 
 
 class EdbSiwave(object):
-    """EdbSiwave class.
+    """EDB Siwave application interface.
 
+    Parameters
+    ----------
+    parent : str
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:259]:
```suggestion
    parent :
```
Since it is an object.

File: pyaedt/edb_core/siwave.py [hunk 8]
```diff
@@ -239,32 +272,32 @@ def __init__(self, parent):
...
     @property
     def _siwave_source(self):
-        """ """
+        """SIwave source."""
         return self.parent.edblib.SIwave.SiwaveSourceMethods
 
     @property
     def _siwave_setup(self):
-        """ """
+        """SI setup."""
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:280]:
```suggestion
        """SIwave setup."""
```

File: pyaedt/edb_core/siwave.py [hunk 9]
```diff
@@ -594,6 +627,18 @@ def add_siwave_dc_analysis(self, accuracy_level=1):
...
+        
+        Parameters
+        ----------
+        source : str
+            Name of the source.
+        
+        Returns
+        -------
+        type
+
```
Comment (jleonatti) [inline pyaedt/edb_core/siwave.py:640]:
```suggestion

```
This method doesn't return anything.

File: pyaedt/modules/LayerStackup.py [hunk 10]
```diff
@@ -1,5 +1,5 @@
 """
-This module contains these classes: ''Layer'' and ''Layers''.  
+This module contains these classes: 'Layer' and 'Layers'.
```
Comment (jleonatti) [inline pyaedt/modules/LayerStackup.py:2]:
```suggestion
This module contains these classes: `Layer` and `Layers`.  
```
You want tick marks here, right?

File: pyaedt/modules/LayerStackup.py [hunk 11]
```diff
@@ -390,11 +390,12 @@ def remove_stackup_layer(self):
 
 
 class Layers(object):
-    """Layers class.
+    """Manages layers for the Circuit and HFSS 3D Layout tools.
     
     Parameters
     ----------
-    parent :
+    parent : str
```
Comment (jleonatti) [inline pyaedt/modules/LayerStackup.py:397]:
```suggestion
    parent :
```

File: pyaedt/modules/Material.py [hunk 12]
```diff
@@ -369,8 +351,8 @@ def add_thermal_modifier_free_form(self, formula, index=0):
 
         Returns
         -------
-        bool
-            ``True`` when successful, ``False`` when failed.
+        type
```
Comment (jleonatti) [inline pyaedt/modules/Material.py:354]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```
The call to `_add_thermal_modifier` ultimately calls `update` on the `Material` class, which returns true or false.

File: pyaedt/modules/Material.py [hunk 13]
```diff
@@ -396,8 +378,8 @@ def add_thermal_modifier_dataset(self, dataset_name, index=0):
...
```
Comment (jleonatti) [inline pyaedt/modules/Material.py:381]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```
Same reasoning as above.

File: pyaedt/modules/Material.py [hunk 14]
```diff
@@ -446,8 +428,8 @@ def add_thermal_modifier_closed_form(self, tref=22, c1=0.0001, c2=1e-6, tl=-273.
...
```
Comment (jleonatti) [inline pyaedt/modules/Material.py:431]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```
Same reason as above.

File: pyaedt/modules/Material.py [hunk 15]
```diff
@@ -643,14 +623,12 @@ def _update_props(self, propname, provpavlue, update_aedt=True):
...
-    """Material class.
-    
-    This class provides all functionalities for material properties.
+    """Manages material properties.
     
     Parameters
     ----------
-    parent : 
-    
+    parent : str
```
Comment (jleonatti) [inline pyaedt/modules/Material.py:630]:
```suggestion
    parent :
```

File: pyaedt/modules/MaterialLib.py [hunk 16]
```diff
@@ -477,6 +477,16 @@ def export_materials_to_file(self, full_json_path):
...
+            """Find datasets.
+            
+            Parameters
+            ----------
+            d :
+            
+            out_list : list
+            
+            """
+
```
Comment (jleonatti) [inline pyaedt/modules/MaterialLib.py:489]:
```suggestion

```
This is a nested method and cannot be accessed from anywhere but inside the `export_materials_to_file` method, so it does not really need documented.

File: pyaedt/siwave.py [hunk 17]
```diff
@@ -1,10 +1,7 @@
 """
-This module initializes Siwave and MessageManager and manages Siwave release and closing.
+This module contains the `siwave` class.
```
Comment (jleonatti) [inline pyaedt/siwave.py:2]:
```suggestion
This module contains the `Siwave` class.  
```

File: pyaedt/siwave.py [hunk 18]
```diff
@@ -1,10 +1,7 @@
 """
-This module initializes Siwave and MessageManager and manages Siwave release and closing.
+This module contains the `siwave` class.  
 
-The ``siwave`` module can be initialized as standalone before launching an app or automatically initialized by an app to the latest installed AEDT version.
-
-
-================
+The `siwave` module can be initialized as standalone before launching an app or automatically initialized by an app to the latest installed AEDT version.
```
Comment (jleonatti) [inline pyaedt/siwave.py:4]:
```suggestion
The `Siwave` module can be initialized as standalone before launching an app or automatically initialized by an app to the latest installed AEDT version.
```

File: pyaedt/siwave.py [hunk 19]
```diff
@@ -41,19 +38,15 @@
 
 
 class Siwave:
-    """
-    This class initializes Siwave based on the inputs provided.
+    """Initializes SIwave based on the inputs provided and manages Siwave release and closing.
```
Comment (jleonatti) [inline pyaedt/siwave.py:41]:
```suggestion
    """Initializes Siwave based on the inputs provided and manages Siwave release and closing.
```

File: pyaedt/siwave.py [hunk 20]
```diff
@@ -144,50 +137,50 @@ def __init__(self, specified_version=None):
...
     @property
     def lock_file(self):
-        """ """
+        """Lock file."""
         return os.path.join(self.project_path, self.project_name + '.siw.lock')
 
     @property
     def results_directory(self):
-        """ """
+        """Results. directory."""
```
Comment (jleonatti) [inline pyaedt/siwave.py:161]:
```suggestion
        """Results directory."""
```

**Review by jleonatti — APPROVED** [review]

**Pull request merged**